### PR TITLE
Add types for @creativebulma/bulma-tagsinput

### PR DESCRIPTION
**Pull request by remcohaszing**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by peterblazejewicz — COMMENTED** [review]
@remcohaszing 
sorry for late review,
what's the reason the class is missing its inherited type and methods (Component/EventEmitter). Can this be detrimental to the intended usage?
Otherwise ready to merge, just this question

**Comment by remcohaszing** [comment]
> what's the reason the class is missing its inherited type and methods and methods

These classes are present in the `src` directory, but they are not accessible in `dist` files. The missing methods are undocumented and used for internal purposes only.

I believe these classes are copied around across similar repositories by the author for convenience.

File: types/creativebulma__bulma-tagsinput/creativebulma__bulma-tagsinput-tests.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
+import BulmaTagsInput, { BulmaTagsInputItem } from 'creativebulma__bulma-tagsinput';
```
Comment (peterblazejewicz) [inline types/creativebulma__bulma-tagsinput/creativebulma__bulma-tagsinput-tests.ts:1]: this should be change after paths are corrected to:
```suggestion
import BulmaTagsInput, { BulmaTagsInputItem } from '@creativebulma/bulma-tagsinput';
```

File: types/creativebulma__bulma-tagsinput/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
```
Comment (peterblazejewicz) [inline types/creativebulma__bulma-tagsinput/tsconfig.json:1]: This config requires paths configuraio for scoped package, see:
https://github.com/definitelyTyped/DefinitelyTyped/#what-about-scoped-packages
Probably it was not created by generator

**Comment by peterblazejewicz** [comment]
It really looks like using inheritance:
https://demo.creativebulma.net/components/tagsinput/1.0/get-started/
```
BulmaTagsInput.prototype
Component {constructor: ƒ, _init: ƒ, _build: ƒ, _bindEvents: ƒ, _caretAtStart: ƒ, …}
```
The dist folder seems to host UMD module and the docs points that `BulmaTagsInput` can be used directly:
https://github.com/CreativeBulma/bulma-tagsinput/blob/84fd693898b49228181d59fb2ec062d631f26142/src/docs/content/get-started/javascript-api.md#L13
So this should compile with proposed types:
```ts
// Instantiate on one element
new BulmaTagsInput('input', {});
// Instantiate on multiple elements
BulmaTagsInput.attach('input', {});
const inputTags =  document.getElementById('input-tags') as HTMLInputElement;
new BulmaTagsInput(inputTags);
// Access to the BulmaTagsInput instance
const btiInstance = inputTags.BulmaTagsInput();
// Call BulmaTagsInput method
btiInstance.add('new Tag');

BulmaTagsInput.attach(
  'input[data-type="tags"], input[type="tags"], select[data-type="tags"], select[type="tags"]',
  {
    allowDuplicates: false,
    caseSensitive: true,
    clearSelectionOnTyping: false,
    closeDropdownOnItemSelect: true,
    delimiter: ',',
    freeInput: true,
    highlightDuplicate: true,
    highlightMatchesString: true,
    itemValue: undefined,
    itemText: undefined,
    maxTags: undefined,
    maxChars: undefined,
    minChars: 1,
    noResultsLabel: 'No results found',
    placeholder: '',
    removable: true,
    searchMinChars: 1,
    searchOn: 'text',
    selectable: true,
    source: undefined,
    tagClass: 'is-rounded',
    trim: true,
  }
);
```
to achieve this different approachs are possible, including one from templates:
```ts
declare namespace BulmaTagsInput {
    
    interface BulmaTagsInputItem {
        ...
    }
    
    interface BulmaTagsInputOptions {
       ...
    }
    
    interface BulmaTagsInputEventMap {
        ...
    }
}

declare class BulmaTagsInput {
    ...
}

export as namespace BulmaTagsInput;

export = BulmaTagsInput;
```
The `HTMLInputTag` could be probably augmented to host that library dedicated func, e.g.:
```ts
interface HTMLInputElement {
    BulmaTagsInput: () => BulmaTagsInput;
}
```
what do you think?

**Comment by remcohaszing** [comment]
Looking at [package.json](https://github.com/CreativeBulma/bulma-tagsinput/blob/v1.0.2/package.json#L11), `src/index.js` is exposed as the main. This is an ESM which exports `BulmaTagsInput` as default. The dist files are unused when using `import '@creativebulma/bulma-tagsinput'.

This does mean the `src` internal files are available, but they are not relevant for usage of the library. `events.js` and `component..js` be typed anyway or is it ok to leave these on the exposed class as-is?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@remcohaszing thx for the contribution!

**Pull request merged**